scripts/microbe_stage/microbe_editor/microbe_editor.as

```diff
@@ -228,7 +227,10 @@ class MicrobeEditor{
 
     private void _addOrganelle(PlacedOrganelle@ organelle)
     {
-        EditorAction@ action = EditorAction(organelle.organelle.mpCost,
+        if((organelle.organelle.needNucleus <= nucleusIsPresent) || (organelle.organelle.needNucleus == 2 && nucleusIsPresent == 0))
```

Comment:
Why this check is like this: `organelle.organelle.needNucleus == 2`?

Reply:
i try to work with Boolean, the problem is the fact that organelle nucleus do not need nucleus, so when i'm trying to put a false organelle if it's a nucleus and nucleus is not present, by change values in int i just check one time. could be made with Boolean but the if statement become bigger.

Reply:
Readable is better. Right now I have no clue reading this that that value being 0 and 2 means.
Better yet would be to split it in half to have like first check `organelle.organelle.needNucleus && !nucleusIsPresent` and then `(organelle.organelle.name == "nucleus") && nucleusIsPresent` that way it is clearer whtat is going on. There should also be a comment saying what those checks are for.